### ISSUE TITLE
 Composer top padding collapsed in room after scroll up

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22389]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to a room
2. add some chats until we can scroll up the screen
3. Scroll to the bottom and scroll upp
## Expected Result: 
There should be some padding at the top of the composer
## Actual Result: 
No padding at the top of the composer, it will be notes when hovering over the message which is closer to compose box

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.37-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/f493e7b7-1160-4116-9a4a-d93d066b80cd

<img width="960" alt="Untitled" src="https://github.com/Expensify/App/assets/43996225/c9b535d5-9090-4b52-b3ac-4c84179fed6f">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @wildan-m
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688604509171109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Composer top padding is not visible.

### What is the root cause of that problem?
- The root cause is that the Participant's Local Time is not visible when in the room thread. But is visible when in Chat thread. Thus showing inconsistency in padding over the composer.

<details>
<summary> Chat Thread </summary>

<img width="668" alt="Screenshot 2023-07-07 at 8 26 35 AM" src="https://github.com/Expensify/App/assets/78416198/5e56349e-7ce6-408f-95da-9ade059ea344">

</details>

<details>
<summary> Room Thread </summary>

<img width="666" alt="Screenshot 2023-07-07 at 8 26 47 AM" src="https://github.com/Expensify/App/assets/78416198/9fc3372e-6272-4363-a0e0-c0c92cd7cd95">

</details>


### What changes do you think we should make in order to solve the problem?
- Solution 1: Add a top padding to the composer by default.
- Solution 2: Render a different component similar to the Participant's Local Time.

### What alternative solutions did you explore? (Optional)
- N/A

**Comment by mkarkachov** [comment]
@kavimuru do we need padding there? with padding it will look cropped anyway while user scrolls messages, check my screenshot with added top  padding

![image](https://github.com/Expensify/App/assets/1360242/26b4b3d3-0077-405d-acab-71cd9b918332)

I disagree with root cause in submitted [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1624580025) because on provided Recording and Screenshot it's chatroom with **Concierge** and 

> Participant's Local Time

 should not be displayed there, same for `#announce` chatroom with many Participants, we can't display local time of all Participants

**Comment by sarious** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Composer top padding collapsed in room after scroll up.

### What is the root cause of that problem?

The root cause of the problem is that there is paddings for scrollable container (chat content). When the user scrolls up the paddings is invisible.

<details>
  <summary>Invisible paddings of chat content</summary>
  
<img width="561" alt="Screenshot 2023-07-07 at 22 42 17" src="https://github.com/Expensify/App/assets/6162207/ee1522eb-0a7d-4f2f-8271-cef6899996eb">
</details>


### What changes do you think we should make in order to solve the problem?
1. So first of all to solve the issue we need to get rid of vertical paddings of chat content. So we need to remove `paddingVertical: 16` from _chatContentScrollView_ in _styles.js_
2. After that we need to add 16px top padding to _ReportActionCompose_ container. But we need to add padding only in case if we don't need to show _Participant Local Time_. So for that we need to add  `{marginTop: shouldShowReportRecipientLocalTime && hasReportRecipient ? 0 : 16}` to the top View in _ReportActionCompose_ component.

### What alternative solutions did you explore? (Optional)

N/A

**Comment by sarious** [comment]
Contributor details
Your Expensify account email: sarious.w@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~012449b41f0680e59b

**Comment by mkarkachov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Margin top is missed at the top of `ReportActionCompose` component

### What is the root cause of that problem?
If it's multi-user chatroom or concierge room `ReportRecipientLocalTime` is not appeared and in this case top margin should be add 

### What changes do you think we should make in order to solve the problem?


https://github.com/Expensify/App/blob/2ec306726201ed7f79d8547a8961c0b41edab271/src/pages/home/report/ReportActionCompose.js#L960
after this line we should add top **margin** (padding will be displayed incorrect) with 
```javascript
!shouldShowReportRecipientLocalTime && styles.mt4
```

https://github.com/Expensify/App/blob/2ec306726201ed7f79d8547a8961c0b41edab271/src/pages/home/report/ReportActionsList.js#L159

here to remove bottom padding in scroll area, we should remove this code
```javascript
shouldShowReportRecipientLocalTime && styles.pt0
```

and in styles.js replace this code 
https://github.com/Expensify/App/blob/2ec306726201ed7f79d8547a8961c0b41edab271/src/styles/styles.js#L1468
with `paddingBottom: 16,`

### What alternative solutions did you explore? (Optional)

N/A

**Comment by twisterdotcom** [comment]
How are these proposals looking @0xmiroslav?

**Comment by jeet-dhandha** [comment]
Any thoughts on my proposal https://github.com/Expensify/App/issues/22389#issuecomment-1624580025 ?

**Comment by nebiyuelias1** [comment]
> ```diff
> paddingTop: shouldShowReportRecipientLocalTime && hasReportRecipient ? 0 : 16
> ```

But did you check out the style guide? It says inline styles are not allowed: https://github.com/Expensify/App/blob/main/contributingGuides/STYLING.md#inline-styles

**Comment by nebiyuelias1** [comment]
Contributor details
Your Expensify account email: nebiyuelias1@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01010c50fbb48ba9b1

**Comment by sarious** [comment]
@nebiyuelias1 Thanks for the comment! This was the initial decision, it can be refactored a little. So, I've refactored my proposal little bit with adding _chatItemWithoutLocalTime_ to _styles.js_.

**Comment by twisterdotcom** [comment]
Bump @0xmiroslav. I'll reassign to another C+ tomorrow if you're a bit busy, don't worry.

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Composer top padding collapsed in the room after scrolling up

### What is the root cause of that problem?
There is no local time for a group, so the `<ParticipantLocalTime>` is completely hidden.

### What changes do you think we should make in order to solve the problem?
We can add a new `isContentVisible` prop to `<ParticipantLocalTime>` to hide only the text without hiding the container style.

change this line:

https://github.com/Expensify/App/blob/cc3761bd00c5b8e32883e47c7ac6547f20698369/src/pages/home/report/ReportActionCompose.js#L994

to

```
                    <ParticipantLocalTime isContentVisible={shouldShowReportRecipientLocalTime && hasReportRecipient} participant={reportRecipient} />
```

And this line:

https://github.com/Expensify/App/blob/cc3761bd00c5b8e32883e47c7ac6547f20698369/src/pages/home/report/ParticipantLocalTime.js#L50-L62

to
```
        <View style={[styles.chatItemComposeSecondaryRow]}>
            {props.isContentVisible && <Text
                style={[styles.chatItemComposeSecondaryRowSubText, styles.chatItemComposeSecondaryRowOffset, styles.pre]}
                numberOfLines={1}
            >
                {translate('reportActionCompose.localTime', {
                    user: reportRecipientDisplayName,
                    time: localTime,
                })
            }
            </Text>}
        </View>
```
If the top padding is too high we can adjust the view height, in my example, we can change the height  to 0 and keep the vertical margin.
```
        <View style={[styles.chatItemComposeSecondaryRow, !props.isContentVisible ? styles.h0 : {}]}>
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by sarious** [comment]
> As this is minor style fix, I'd like to pick up the most accurate and simple solution which doesn't cause any minor regressions. No satisfactory proposals yet.

Please, see my [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1625914017). You may have thought that there is some kind of complicated solution, because a lot has been written. But it seems, it will be difficult to find more accurate and simple solution. We only get rid of `paddingVertical: 16` in one place and add `!isChatItemWithLocalTime && styles.pt4` to other View. 
Also I've done small refactoring with extracting new variable:
`const isChatItemWithLocalTime = shouldShowReportRecipientLocalTime && hasReportRecipient;`

So the simple fix without refactoring will looks in this way, but I think it's better to extract new `isChatItemWithLocalTime` variable as in my proposal:
```diff
chatContentScrollView: {
    flexGrow: 1,
    justifyContent: 'flex-start',
-   paddingVertical: 16,
},
```

```diff
            <View
                style={[
                    shouldShowReportRecipientLocalTime && !lodashGet(this.props.network, 'isOffline') && styles.chatItemComposeWithFirstRow,
                    this.props.isComposerFullSize && styles.chatItemFullComposeRow,
+                  !(shouldShowReportRecipientLocalTime && hasReportRecipient) && styles.pt4,
                ]}

```

**Comment by sarious** [comment]
@0xmiroslav Oh, i see. It will be better not removing `paddingVertical`, but replacing `paddingVertical` with `paddingBottom`. So there will be no issues, because of `paddingBottom`. This variant will be better:
<img width="611" alt="image" src="https://github.com/Expensify/App/assets/6162207/0a84d402-cbf2-4493-a66b-b4f6be436dfd">

**Comment by wildan-m** [comment]
@0xmiroslav thoughts on my main and alternative proposal?

**Comment by sarious** [comment]
> @sarious please update your proposal. Remove all code diffs which are not allowed in proposal stage.

I've updated [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1625914017), removed code diffs.

**Comment by twisterdotcom** [comment]
Bump on this @0xmiroslav.

**Comment by twisterdotcom** [comment]
@0xmiroslav please let me know as soon as you can whether you can review these again. I'll reassign by Thursday otherwise.

**Comment by situchan** [comment]
I'd like to C+ review this as I already reviewed [related issue](https://github.com/Expensify/App/issues/23187).

**Comment by twisterdotcom** [comment]
Eugh, automation is great when it works! Alas. Okay @situchan - assigned you as the C+.

**Comment by twisterdotcom** [comment]
@situchan you stepped in here, but we're still waiting on your reviews for the two proposals in here please.

**Comment by mkhutornyi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Composer top padding collapsed in the room after scrolling up

### What is the root cause of that problem?
There's 16px padding when show local time or hide composer but this is applied to ScrollView's `contentContainerStyle` so padding part is also scrollable.
https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/pages/home/report/ReportActionsList.js#L179

### What changes do you think we should make in order to solve the problem?
We should move padding outside of scrollable view.
1. update `chatContentScrollView` style: `paddingVertical: 16` => `paddingBottom: 16`
2. remove `shouldShowReportRecipientLocalTime || shouldOmitBottomSpace ? styles.pt0 : {}` in [contentContainerStyle](https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/pages/home/report/ReportActionsList.js#L179C17-L179C38)
3. add `shouldShowReportRecipientLocalTime || hideComposer ? {} : styles.pb4` to container View [style](https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/pages/home/report/ReportActionsList.js#L173)
4. To be responsive to archived rooms, anonymous user, public rooms where comment is not allowed, add conditional 16px top margin [here](https://github.com/Expensify/App/blob/024d210bc0b5a0454b62a72de60057af67cdd045/src/pages/home/report/ReportFooter.js#L71): `isArchivedRoom || isAnonymousUser ? styles.mt4 : {}`

**Comment by wildan-m** [comment]
@situchan have you tried the alternative solution?

**Comment by situchan** [comment]
@wildan-m still not responsive to ArchivedReportFooter, AnonymousReportFooter

**Comment by mkarkachov** [comment]
@situchan did you have a chance to check my proposal? https://github.com/Expensify/App/issues/22389#issuecomment-1625970966

**Comment by situchan** [comment]
@mkarkachov Based on https://github.com/Expensify/App/issues/22389#issuecomment-1638602628, I think your proposal was already reviewed.
If I re-review, your solution still doesn't meet https://github.com/Expensify/App/issues/22389#issuecomment-1658039910.

**Comment by mkarkachov** [comment]
Not sure that proposals before that comment were reviewed but ok

**Comment by situchan** [comment]
@mkarkachov's [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1657936582) looks good to me.
It's the only proposal which covers all possible cases:
- composer shown / hidden
- archived / not archived
- logged in / anonymous

🎀 👀 🎀 C+ reviewed

**Comment by marcaaron** [comment]
Sounds good. Thank you guys for the work so far. Let's open the PR 🚀

**Comment by mkarkachov** [comment]
I will start on PR tomorrow morning and will complete it during a day

**Comment by situchan** [comment]
Oh sorry @marcaaron, I meant @mkhutornyi
The name was similar and I accidentally tagged wrong name, though I pointed correct proposal [link](https://github.com/Expensify/App/issues/22389#issuecomment-1657936582).
I do apologize for everyone.

**Comment by marcaaron** [comment]
No worries. I missed that the linked proposal and name did not match up. 

Is there any reason why @mkarkachov can't do that proposal? Not sure what else to say other than... please be more careful. I'd rather not take an opportunity away from @mkarkachov because you meant to recommend someone else. 🤷

**Comment by situchan** [comment]
> Is there any reason why @mkarkachov can't do that proposal? Not sure what else to say other than... please be more careful. I'd rather not take an opportunity away from @mkarkachov because you meant to recommend someone else. 🤷

I provided feedback [here](https://github.com/Expensify/App/issues/22389#issuecomment-1658478191) so @mkarkachov's solution still doesn't cover these cases:


<img width="1485" alt="Screenshot 2023-08-01 at 9 24 43 AM" src="https://github.com/Expensify/App/assets/108292595/8ce16ddf-197a-4d09-9e64-7de855a1a489">

<img width="1090" alt="Screenshot 2023-08-01 at 9 26 15 AM" src="https://github.com/Expensify/App/assets/108292595/f390ec39-1d78-4459-aefa-0827ee64d665">

**Comment by sarious** [comment]
> @sarious your solution doesn't work when composer is hidden

It's not difficult to fix my solution. We just need to add `paddingTop: 16` to container with ArchivedReportFooter. We can do it by adding `styles.pt4` to wrapper View in `ReportFooter.js`:
<img width="710" alt="image" src="https://github.com/Expensify/App/assets/6162207/50749e91-c0e5-473d-a9ae-aa06647cc438">

So I've updated my [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1625914017), please review it.

**Comment by situchan** [comment]
I already [approved](https://github.com/Expensify/App/issues/22389#issuecomment-1658757653) the first completed proposal.
These kinds of things can be easily missed even during PR review. No one has ever thought of archived footer, anonymous footer cases before @mkhutornyi's [proposal](https://github.com/Expensify/App/issues/22389#issuecomment-1657936582) so it's not true that they could be tackled in PR.

**Comment by mkarkachov** [comment]
> it's not true that they could be tackled in PR.

@situchan We would pay attention in PR to it anyway. I posted my proposal a month ago and [#21950 ](https://github.com/Expensify/App/pull/21950) was merged on July 25 to main branch with [code](https://github.com/mkarkachov/App/commit/50a63158c36a791fc6eb052068da989c3756a118#diff-9ac76d645ad53156c86195c1a8f5adb3bb9b71f78351e43c4e1b6d6896921c4eR170-R172)

```javascript
const isArchivedRoom = ReportUtils.isArchivedRoom(props.report);
const hideComposer = ReportUtils.shouldHideComposer(props.report, errors);
const shouldOmitBottomSpace = hideComposer || isArchivedRoom;
```

It's hard to misunderstand and ignore names of those constants :)

**Comment by mkarkachov** [comment]
@sarious Uploaded last videos for iOS, Android and Mobile Web - Chrome. PR ready for review #23992

**Comment by situchan** [comment]
@mkarkachov I tested PR. It really causes [this regression](https://github.com/Expensify/App/issues/22389#issuecomment-1660295060).

Another regression:

<img width="942" alt="Screenshot 2023-08-01 at 8 13 26 PM" src="https://github.com/Expensify/App/assets/108292595/e4b62bfb-5718-4333-a7db-da7fa70c876a">

**Comment by situchan** [comment]
@marcaaron personally I think no-regression is important factor when accept proposal. No one wants to get penalty because of these minor regressions.
I haven't found any regressions on @mkhutornyi's [solution](https://github.com/Expensify/App/issues/22389#issuecomment-1657936582).

**Comment by mkarkachov** [comment]
@situchan it's normal process for any PR, please pay more attention to your own responsibilities next time to avoid penalty. Can we get fast final decision in this case? I don't want to waste my time, should I apply updates to my PR or will anyone assign this issue to another guy? Who has the last word? @twisterdotcom @marcaaron

**Comment by mkarkachov** [comment]
> @mkarkachov I tested PR. It really causes [this regression](https://github.com/Expensify/App/issues/22389#issuecomment-1660295060).
> 
> Another regression:
> <img alt="Screenshot 2023-08-01 at 8 13 26 PM" width="942" src="https://user-images.githubusercontent.com/108292595/257542069-e4b62bfb-5718-4333-a7db-da7fa70c876a.png">

@situchan I made screencast. In my PR there is not regression with hidden compose field, in case if admin restrict posting to another users. Check it please

https://github.com/Expensify/App/assets/1360242/bc3f9173-913d-4e5a-b856-f413ce50c9cc

**Comment by marcaaron** [comment]
Discussing internally.

**Comment by mkarkachov** [comment]
Just in case: [contribution guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 

![Screenshot_20230801_221643_Firefox](https://github.com/Expensify/App/assets/1360242/d5fc9661-1460-49d3-b24a-ab164d6dfb1f)

**Comment by mkarkachov** [comment]
Sounds great. @mkhutornyi @situchan we still have time to get a bonus ;) please review my PR. I applied fixes to avoid regression issues mentioned above.

**Comment by situchan** [comment]
As per https://github.com/Expensify/App/issues/22389#issuecomment-1661061391, @mkhutornyi you can also help testing [PR](https://github.com/Expensify/App/pull/23992) and find any possible regressions.

**Comment by twisterdotcom** [comment]
This isn't overdue. We're waiting for https://github.com/Expensify/App/pull/23992 to hit prod for 7 days.

**Comment by twisterdotcom** [comment]
Payment summary:

@mkarkachov paid $1500 here: https://www.upwork.com/nx/wm/offer/25865850 (Contributor)
@situchan paid $1500 here: https://www.upwork.com/nx/wm/offer/25763008) (Contributor+)
@wildan-m never applied: https://github.com/Expensify/App/issues/22389#issuecomment-1661061745. Due $250 if they do.

@situchan please complete the checklist.

**Comment by situchan** [comment]
- [x] The PR that introduced the bug has been identified. Link to the PR: N/A
- [x] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] Determine if we should create a regression test for this bug.
- [x] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal:**
There are many test cases to consider as stated in [PR](https://github.com/Expensify/App/pull/23992) description
So in summary, this can be simplified test step:
1. Go to any chat room with enough messages history
2. While scrolling messages, verify that there's always some padding above composer

**Comment by wildan-m** [comment]
@twisterdotcom I thought Melvin stored my UW account. Just applied

**Comment by twisterdotcom** [comment]
Ahh yes. @mkhutornyi could you apply for the job as well? [Upwork job](https://www.upwork.com/jobs/~016810fbd90ab7373e)

**Comment by twisterdotcom** [comment]
@wildan-m you've been paid.
@mkarkachov please accept the offer here: https://www.upwork.com/nx/wm/offer/26194971

**Comment by twisterdotcom** [comment]
@CortneyOfstad I'm out for Parental Leave. Please:
1. Ask @situchan to complete the checklist.
2. Pay @mkarkachov $1000 + $500 bonus here: https://www.upwork.com/nx/wm/offer/26194971

**Comment by CortneyOfstad** [comment]
@situchan We need you to complete the check list [here](https://github.com/Expensify/App/issues/22389#issuecomment-1670895408) and link the appropriate details. Thank you!

**Comment by situchan** [comment]
> @situchan We need you to complete the check list [here](https://github.com/Expensify/App/issues/22389#issuecomment-1670895408) and link the appropriate details. Thank you!

Here: https://github.com/Expensify/App/issues/22389#issuecomment-1680809342

**Comment by CortneyOfstad** [comment]
Sorry for the confusion! We ask that you work in the original checklist, otherwise the GH cannot be closed. Thank you!
